src/RoutingList/RoutingListContainer.js

```diff
@@ -0,0 +1,122 @@
+import {
+  useCallback,
+  useState,
+} from 'react';
+import { omit } from 'lodash';
+import { FormattedMessage } from 'react-intl';
+import {
+  useHistory,
+  useParams,
+} from 'react-router';
+
+import { useShowCallout } from '@folio/stripes-acq-components';
+import {
+  ConfirmationModal,
+  LoadingView,
+} from '@folio/stripes/components';
+
+import { UNIQUE_NAME_ERROR_CODE } from './constants';
+import {
+  useRoutingListById,
+  useRoutingListMutation,
+} from './hooks';
+import RoutingListForm from './RoutingListForm';
+
+import css from './RoutingList.css';
+
+export const RoutingListContainer = () => {
+  const showCallout = useShowCallout();
+  const history = useHistory();
+  const { id, poLineId } = useParams();
+  const { routingList, isLoading } = useRoutingListById(id);
+  const { deleteListing, createListing, updateListing } = useRoutingListMutation();
+
+  const [confirmDelete, setConfirmDelete] = useState(false);
+  const showConfirmDelete = useCallback(() => setConfirmDelete(true), []);
+  const hideConfirmDelete = useCallback(() => setConfirmDelete(false), []);
+
+  const onClose = useCallback(() => {
+    history.goBack();
+  }, [history]);
+
+  const onMutationSuccess = useCallback((messageId = 'ui-orders.routing.list.update.success') => {
+    onClose();
+    showCallout({ messageId });
+  }, [onClose, showCallout]);
+
+  const onMutationError = useCallback((messageId = 'ui-orders.routing.list.update.error') => {
+    showCallout({
+      messageId,
+      type: 'error',
+    });
+  }, [showCallout]);
+
+  const onDelete = async () => {
+    hideConfirmDelete();
+    await deleteListing(
+      routingList.id,
+      {
+        onSuccess: () => onMutationSuccess('ui-orders.routing.list.delete.success'),
+        onError: () => onMutationError('ui-orders.routing.list.delete.error'),
+      },
+    );
+  };
+
+  const onSubmit = async (values) => {
+    const dataToSave = omit(values, ['users']);
+
+    if (routingList?.id) {
```

Comment:
do not create such functions, when you can create 2 containers for form component ListCreate and ListEdit